### DISCUSSION TITLE
Intelligent/automated meal planning

### DISCUSSION THREAD

**Discussion by spieker** [mealie-recipes/mealie#892] [Feature Request]
**Before submitting this feature request I have**

- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**

Having to cook for the family every week and taking under consideration the schedules for all children can be "complex". Let's say you want to eat salat two days a week. Two days the meal must be prepared, since the kids are coming home late. Creating a meal plan simplifies this process a lot. However, in my experience, eventually you forget to create one for the next week, and at some point you stop creating one at all. Having a meal plan that is creating itself and takes some rules under consideration, would let you come back to using the meal plan at any point in time, even if you have stopped for some days/weeks/months.

**(Optional): Suggest A Solution**

It would be great to be able to define auto assignment rules for meal planning. These rules could be based on categories (like now), plus certain categories/tags for each day of the week. Meals could then be tagged with "preparable", "salad", "casserole", "soup", ... and a meal plan could be created automatically some time in advance (i.e. always one week in advance, or a configurable amount of days in advance). The rules could then maybe read:

```
Only use recipes from the category "Main dish"

* Monday: must be tagged with "casserole"
* Tuesday: anything (no further rules)
* Wednesday: must be tagged with "soup"
* Thursday: must be tagged with "salad"
* Friday: must be tagged with "salad"
```

**Additional Information**

- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by hay-kot** [comment]
Please use the feature request template. See below.

- https://github.com/hay-kot/mealie/issues/317

**Reply by spieker** [reply]
Done. Please let me know if you need something else

**Comment by hay-kot** [comment]
I like it! Tracking an issue/tasks here:

- https://github.com/hay-kot/mealie/issues/913

I probably won't be able to get to it for a while, but it's there!

**Comment by hay-kot** [comment]
Meal-plan rules can now be set - check it out 

https://beta.mealie.io/group/mealplan/settings
